packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,350 @@
+
+    function _requestOraclePriceRelay(
+        address l1Token,
+        uint256 amount,
+        uint256 requestTimestamp,
+        bytes memory customAncillaryData,
+        uint256 proposerRewardPct
+    ) private {
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+        uint256 proposerBondPct =
+            FixedPoint.Unsigned(bridgePoolFactory.getProposerBondPct()).div(FixedPoint.fromUnscaledUint(1)).rawValue;
+
+        // Optimistic oracle will pull proposer reward from passive LP pool.
+        uint256 proposerReward =
+            FixedPoint
+                .Unsigned(proposerRewardPct)
+                .div(FixedPoint.fromUnscaledUint(1))
+                .mul(FixedPoint.Unsigned(amount))
+                .rawValue;
+
+        IERC20(l1Token).safeApprove(address(optimisticOracle), proposerReward);
+        optimisticOracle.requestPrice(
+            bridgePoolFactory.getIdentifier(),
+            requestTimestamp,
+            customAncillaryData,
+            IERC20(l1Token),
+            proposerReward
+        );
+
+        // Set the Optimistic oracle liveness for the price request.
+        optimisticOracle.setCustomLiveness(
+            bridgePoolFactory.getIdentifier(),
+            requestTimestamp,
+            customAncillaryData,
+            bridgePoolFactory.getOptimisticOracleLiveness()
+        );
+
+        // Set the Optimistic oracle proposer bond for the price request.
+        uint256 proposerBond = FixedPoint.Unsigned(proposerBondPct).mul(FixedPoint.Unsigned(amount)).rawValue;
+        optimisticOracle.setBond(
+            bridgePoolFactory.getIdentifier(),
+            requestTimestamp,
+            customAncillaryData,
+            proposerBond
+        );
+    }
+
+    function _proposeOraclePriceRelay(
+        address l1Token,
+        uint256 amount,
+        uint256 requestTimestamp,
+        bytes memory customAncillaryData
+    ) private {
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+        uint256 proposerBondPct =
+            FixedPoint.Unsigned(bridgePoolFactory.getProposerBondPct()).div(FixedPoint.fromUnscaledUint(1)).rawValue;
+        uint256 finalFee = _getStore().computeFinalFee(address(l1Token)).rawValue;
+
+        uint256 totalBond =
+            FixedPoint
+                .Unsigned(proposerBondPct)
+                .mul(FixedPoint.Unsigned(amount))
+                .add(FixedPoint.Unsigned(finalFee))
+                .rawValue;
+
+        // Pull the total bond from the caller so that the OptimisticOracle can subsequently pull it from here.
+        IERC20(l1Token).safeTransferFrom(msg.sender, address(this), totalBond);
+        IERC20(l1Token).safeApprove(address(optimisticOracle), totalBond);
+        optimisticOracle.proposePriceFor(
+            msg.sender,
+            address(this),
+            bridgePoolFactory.getIdentifier(),
+            requestTimestamp,
+            customAncillaryData,
+            1e18
```

Comment:
anything non-zero would work as "true". would `1` be better than `1e18`?

Reply:
I guess that `e1e18` might be easier from a voter perspective though as this is `1` encoded by the standard convention.

Reply:
> I guess that e1e18 might be easier from a voter perspective though as this is 1 encoded by the standard convention.

Yeah that's what I was thinking. I wanted to be consistent with how Admin proposals are currently voted on.

Reply:
@mrice32 what do you think here?

Reply:
Agree that 1e18 is the standard "true" value.